### PR TITLE
User/raiyan/recruit training

### PR DESCRIPTION
**Pull request by RaiyanK23**

### CODE REVIEW

File: Recruit-Training/Basic-Recruit-Training/person/Person.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#include "Person.h"
+
+#include <iostream>
+
+Person::Person()
+{
+  this -> name_ = "";
```
Comment (liconway) [inline Recruit-Training/Basic-Recruit-Training/person/Person.cpp:7]: This goes for all pointers in this PR, but I think `this->` isn't really necessary as it's generally implied.

We generally use the `_` at the end of class data members (see Google's [Variable Names guide](https://google.github.io/styleguide/cppguide.html#Variable_Names)) to signify that it's part of the class, as it's a lot shorter then `this->` and generally makes things easier.

File: Recruit-Training/Basic-Recruit-Training/pointers/main.cpp [hunk 2]
```diff
@@ -19,32 +20,41 @@ int main()
     int *p = &x;
 
     //TODO in the line below predict what what is going to be output
-    cout << "Predicted value of p: " /* << PLACE YOUR PREDICTION HERE*/ << endl;
+    cout << "Predicted value of p: " /* << This will print the address of p/ << endl;
```
Comment (liconway) [inline Recruit-Training/Basic-Recruit-Training/pointers/main.cpp:23]: Two things about this:
1) No need to keep any of your predictions as comments, just print them to console
2) This line will cause your program not to compile, as your comment doesn't end (`*/` vs `/`)

File: Recruit-Training/Basic-Recruit-Training/person/main.cpp [hunk 3]
```diff
@@ -20,14 +20,19 @@ void compareAnswers(const std::string myAnswer, const std::string correctAnswer)
 // Swaps a & b.
 void mySwap(int& a, int& b)
 {
+  int temp = 0;
+  temp = a;
```
Comment (liconway) [inline Recruit-Training/Basic-Recruit-Training/person/main.cpp:24]: For both swap functions, you might as well save a step:
```
int temp = *a;
```

**Review by liconway — CHANGES_REQUESTED** [review]

File: Recruit-Training/Basic-Recruit-Training/person/Person.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  this -> age_ = new int(0);
+}
+
+Person::Person (const std::string& name, int age)
+{
+  this -> name_ = name;
+  this -> age_ = new int(age);
```
Comment (liconway) [inline Recruit-Training/Basic-Recruit-Training/person/Person.cpp:14]: We generally prefer using [Constructor Initializer Lists](https://google.github.io/styleguide/cppguide.html#Constructor_Initializer_Lists)

**Review by liconway — CHANGES_REQUESTED** [review]

**Review by JessieG123 — APPROVED** [review]
Looks good to me! Don't close the PR yet, wait for one more approval and then you can close it.

File: Recruit-Training/Basic-Recruit-Training/person/Person.cpp (hunk 4 above)
Comment (RaiyanK23) [reply Recruit-Training/Basic-Recruit-Training/person/Person.cpp]: Hey Liam! Made the changes as you've requested, please feel free to take a look when you can thanks.

**Review by liconway — APPROVED** [review]
Nice job! 🎉 Congratulations on finishing the basic training! Be sure to ***close*** the PR, do ***not*** merge it.

image

**Pull request closed without merging**